### PR TITLE
Add article model and routes

### PR DESCRIPTION
**Pull request by Drapegnik**

### CODE REVIEW

File: src/api/article/models.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import mongoose, { Schema } from 'mongoose';
+import User from 'api/user';
+
+const ArticleTypeSchema = new Schema({
+  type: {
```
Comment (Drapegnik) [inline src/api/article/models.js]: i think in this case will be better name `name`

File: src/api/article/models.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import mongoose, { Schema } from 'mongoose';
+import User from 'api/user';
+
+const ArticleTypeSchema = new Schema({
```
Comment (Drapegnik) [inline src/api/article/models.js]: If you want can separate models by different files: `article.model.js` & `type.model.js`

File: src/api/article/models.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    ref: ArticleType,
+  },
+  slug: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now(),
```
Comment (Drapegnik) [inline src/api/article/models.js]: **BUG!!!** you always will get the same value of one call `Date.now()`, you must give a `function` to `default` and in this case it'll be called on create

File: src/api/article/index.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import { requireAuth } from 'auth';
+import * as controller from './controllers';
+import Article, { ArticleType } from './models';
+
+const router = Router();
+router.get('/', controller.getAll);
+router.get('/:id', controller.getById);
+
+router.post('/', requireAuth, controller.create);
```
Comment (Drapegnik) [inline src/api/article/index.js]: should be protected with `requireAuth` and `requireAdminRole` or smthg with same functionality (modify articles can only `admins` and `creators`)

File: src/api/article/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+router.put('/:id', requireAuth, controller.update);
+
+router.delete('/:id', requireAuth, controller.remove);
```
Comment (Drapegnik) [inline src/api/article/index.js]: and for this routes same as about

**Comment by Drapegnik** [comment]
@asntr update postman api, add test and check that `travis` build success

File: src/api/article/controllers.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+export const getAll = async (req, res, next) => {
+  const page = parseInt(req.query.page, 10) || 0;
+  const pageSize = parseInt(req.query.pageSize, 10) || 10;
+  try {
+    const query = Article.find({}).skip(page * pageSize).limit(pageSize);
+    const articles = await query.exec();
+    res.status(200).send({
+      result: articles,
+      next: {
```
Comment (Drapegnik) [inline src/api/article/controllers.js]: 👍 good work!

File: src/api/article/models.js (hunk 3 above)
Comment (asntr) [inline src/api/article/models.js]: `default: Date.now` will work?

Comment (Drapegnik) [inline src/api/article/models.js]: yep

**Comment by asntr** [comment]
but-it-is-not-this-day-there-may-come-a-day-when-ill-update-postman-api-but-it-is-not-this-day

File: src/auth/index.js [hunk 7]
```diff
@@ -13,6 +13,16 @@ const requireAuth = (req, res, next) => {
   return next();
 };
 
+function requireRoles(roles) {
```
Comment (Drapegnik) [inline src/auth/index.js]: `require` sounds like **all** roles should be (like `a and b`)
mb `allowRoles` (sounds like `one of (a and b)`),

File: src/api/article/controller.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  const pageSize = parseInt(req.query.pageSize, 10) || 10;
+  const skip = page * pageSize;
+  let result;
+
+  return Article
+    .find({})
+    .populate('type')
+    .skip(skip)
+    .limit(pageSize)
+    .then(articles => articles.map(serializeArticle))
```
Comment (asntr) [inline src/api/article/controller.js]: Can't we solve it with smth similar to `query.select('-_id -__v');`?

Comment (Drapegnik) [inline src/api/article/controller.js]: I try this, but there are several problems
1. duplication, we can reuse it only as `export string` with `Article` fields, WTF??
2. select doesn't work with `populate` (for me)

File: src/api/article/tests.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      expect(result).to.be.null; // eslint-disable-line no-unused-expressions
+    });
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
+    it('should work without error', () => {
+      expect(err).to.be.a(undefined);
+    });
+  } finally {
+    await ArticleType.remove({ name: 'test' });
```
Comment (Drapegnik) [inline src/api/article/tests.js]: @asntr for db setup & cleanup you can use [`mocha hooks`](https://mochajs.org/#hooks)

**Comment by Drapegnik** [comment]
[articles model + api](https://trello.com/c/Q4YSYOZW/32-articles-model-api)

**Pull request merged**